filebeat/prospector/log/prospector.go

```diff
@@ -264,10 +267,55 @@ func (p *Prospector) matchesFile(filePath string) bool {
 	return false
 }
 
+type FileSortInfo struct {
+	info os.FileInfo
+	path string
+}
+
+func getSortInfos(paths map[string]os.FileInfo) []FileSortInfo {
+
+	sortInfos := make([]FileSortInfo, 0, len(paths))
+	for path, info := range paths {
+		sortInfo := FileSortInfo{info: info, path: path}
+		sortInfos = append(sortInfos, sortInfo)
+	}
+
+	return sortInfos
+}
+
+func getSortedFiles(scanOrder string, scanSort string, sortInfos []FileSortInfo) []FileSortInfo {
+	switch scanOrder + "_" + scanSort {
+	case "asc_modtime":
+		sort.Slice(sortInfos, func(i, j int) bool {
+			return sortInfos[i].info.ModTime().Before(sortInfos[j].info.ModTime())
+		})
+	case "desc_modtime":
+		sort.Slice(sortInfos, func(i, j int) bool {
+			return sortInfos[i].info.ModTime().After(sortInfos[j].info.ModTime())
+		})
+	case "asc_filename":
+		sort.Slice(sortInfos, func(i, j int) bool {
+			return strings.Compare(sortInfos[i].info.Name(), sortInfos[j].info.Name()) < 0
+		})
+	case "desc_filename":
+		sort.Slice(sortInfos, func(i, j int) bool {
+			return strings.Compare(sortInfos[i].info.Name(), sortInfos[j].info.Name()) > 0
+		})
+	default:
+	}
+
+	return sortInfos
+}
+
 // Scan starts a scanGlob for each provided path/glob
 func (p *Prospector) scan() {
 
-	for path, info := range p.getFiles() {
+	paths := p.getFiles()
+	files := getSortInfos(paths)
```

Comment:
This should not happen in every call, only if sorting is enabled as this could have a performance impact. I would expect something like:

`if config.sort.Enabled() { paths := paths.Sort() `

This is heavily simplified but you get what I'm trying to explain? Do we even need to expose the fileinfo outside of the sort?